lib/src/main/java/graphql/nadel/instrumentation/NadelInstrumentation.kt

```diff
@@ -100,4 +101,14 @@ interface NadelInstrumentation {
     ): CompletableFuture<ExecutionResult> {
         return CompletableFuture.completedFuture(executionResult)
     }
+
+    /**
+     * Called when Nadel encounters an error that can be useful for the caller code - for logging or metrics, for example.
+     *
+     * The nature of the error can be obtained by looking at the value of [NadelInstrumentationOnErrorParameters.errorType].
+     *
+     *  @param parameters to this step
+     */
+    fun onError(parameters: NadelInstrumentationOnErrorParameters) {
```

Comment:
Maybe by default it could have done the same behavior as before. But not important, since we're definitely going to add custom behavior.